### PR TITLE
Implement application bootstrap component

### LINKED ISSUE #4707: Application Bootstrap Loading State
### Description
At very beginning, mounting application wallet need to have loading state in order to fetch default application and create an instance of client with default application

### Acceptance Criteria
- Create a Bootstrap Component and Wrap entire routing with bootstrap component
- Bootstrap should only render Props.children
- In Loading State, children shouldn't mount till loading is over
- In case apis get failed, it should show retry stage and user are not allowed to pass this stage
- Add Unit Test

### PR DESCRIPTION
**Pull request by eniolam1000752**
### What was the problem?

This PR resolves #4707 and #4708

### How was it solved?

- [x] Implement application bootstrap wrapper
- [x] update order of query hook call (i.e app meta data first then applications on-chain data)
- [x] update unit tests

### How was it tested?

- Jest
- Visually

### CODE REVIEW

File: setup/config/webpack.config.dev.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ const config = {
     new webpack.DefinePlugin({
```
Comment (oskarleonard) [inline setup/config/webpack.config.dev.js]: **Comment not related to this file**

Question: This seems to be unrelated to hardware wallet? Shouldnt we merge this into development?

File: src/modules/blockchainApplication/manage/store/reducer.js [hunk 2]
```diff
@@ -36,20 +36,32 @@ export const pins = (state = initialState.pins, { type, chainId }) => {
 /**
  *
  * @param {Object} state
- * @param {type: String, data: Object} action
+ * @param {type: String, app: Object, apps: Object, chainId: String} action
  */
-export const applications = (state = initialState.applications, { type, application, chainId }) => {
+export const applications = (state = initialState.applications, { type, app, apps, chainId }) => {
```
Comment (oskarleonard) [inline src/modules/blockchainApplication/manage/store/reducer.js:41]: It is not advisable [to mutate the redux state ](https://redux.js.org/faq/immutable-data#why-does-a-reducer-mutating-the-state-prevent-react-redux-from-re-rendering-a-wrapped-component) this way. This whole reducer smells a bit to me, dont we have any noticeable bugs on the components utilising this state? 🤔

File: src/modules/blockchainApplication/manage/store/reducer.js [hunk 3]
```diff
@@ -36,20 +36,32 @@ export const pins = (state = initialState.pins, { type, chainId }) => {
...
+      if (app.chainID in state) {
+        state[app.chainID].serviceURLs.push(app.serviceURLs);
       } else {
-        state[application.chainID] = application;
+        state[app.chainID] = app;
       }
       return state;
 
+    case actionTypes.setApplications: {
+      apps.forEach((application) => {
```
Comment (oskarleonard) [inline src/modules/blockchainApplication/manage/store/reducer.js]: Couldnt this be written in an immutable way with just a `.map(`?

File: src/modules/blockchainApplication/manage/store/reducer.js (hunk 2 above)
Comment (eniolam1000752) [reply src/modules/blockchainApplication/manage/store/reducer.js:41]: hmmm nope we don't so we have refactored reducer handlers and all suitable actions to adjust to this

File: src/modules/blockchainApplication/manage/store/reducer.js (hunk 3 above)
Comment (eniolam1000752) [reply src/modules/blockchainApplication/manage/store/reducer.js]: I guess so but I think just iterating through the array list and re-consuming the checks that are done when setting a single application was way easier.

File: setup/config/webpack.config.dev.js [hunk 4]
```diff
@@ -27,6 +27,7 @@ const config = {
...
       PRODUCTION: JSON.stringify(false),
       VERSION: JSON.stringify(version),
+      DEFAULT_NETWORK: 'devnet',
```
Comment (soroushm) [inline setup/config/webpack.config.dev.js]: no iots wront its should be in .env
and need a prefix of REACT_APP like the REACT_APP_MSW you can find in config file
REACT_APP_ DEFAULT_NETWORK

File: setup/react/app/ApplicationBootstrap.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  useEffect(() => {
+    if (defaultApps.length && isFetched) {
+      setCurrentApplication(defaultApps[0]);
+      setApplications(defaultApps);
+    }
+  }, [isFetched]);
+
+  if (error) {
+    // @TODO: this return should be replaced with an actual error message page
```
Comment (soroushm) [inline setup/react/app/ApplicationBootstrap.js:31]: create a ticket if is not exist and add ticket number to todo

File: src/modules/blockchainApplication/manage/store/reducer.js (hunk 3 above)
Comment (soroushm) [reply src/modules/blockchainApplication/manage/store/reducer.js]: @oskarleonard good catch 

@eniolam1000752 redux is imotable we shoulnt manipulate the state in this way we need to return a new object like below

```js
 if (app.chainID in state) {
cons newState = {
 ...state
}
newState[app.chainID].serviceURLs.push(app.serviceURLs)
       return newState
      } else {
       return {
        ...state,
        [app.chainID]: app;
        }
      }
```

Comment (soroushm) [reply src/modules/blockchainApplication/manage/store/reducer.js]: more info here
https://stackoverflow.com/questions/43759080/what-could-happen-if-modifying-state-directly-inside-a-redux-reducer

**Review by soroushm — CHANGES_REQUESTED** [review]

File: setup/config/webpack.config.dev.js (hunk 1 above)
Comment (soroushm) [reply setup/config/webpack.config.dev.js]: @eniola its wrong env you can have .env file in your pc for test propose check this out
https://create-react-app.dev/docs/adding-custom-environment-variables/#what-other-env-files-can-be-used

**Review by soroushm — DISMISSED** [review]

**Review by oskarleonard — DISMISSED** [review]

**Review by soroushm — APPROVED** [review]

**Review by oskarleonard — APPROVED** [review]
